internal/npm_install/generate_pnp_build_files.ts

```diff
@@ -1,44 +1,47 @@
-import * as fs from "fs";
+import { promises as fsp, existsSync, mkdirSync } from "fs";
 import * as path from "path";
 
-const workspacePath = process.argv.slice(2)[0];
-
-function mkdirp(p: string) {
-  if (!fs.existsSync(p)) {
-    mkdirp(path.dirname(p));
-    fs.mkdirSync(p);
-  }
-}
-
 function writeFileSync(p: string, content: string) {
-  mkdirp(path.dirname(p));
-  fs.writeFileSync(p, content);
+  if (!existsSync(path.dirname(p))) {
+    mkdirSync(path.dirname(p), { recursive: true });
+  }
+  return fsp.writeFile(p, content);
```

Comment:
this will return a Promise object now. Which is quite confusing because you use the sync methods above.

I suggest at least changing the signature to:
`function writeFile(p: string, content: string): Promise<void> {`

or even better:

```ts
async function writeFile(p: string, content: string): Promise<void> {
  await fsp.mkdir(path.dirname(p), { recursive: true });
  return await fsp.writeFile(p, content);
}
```

Reply:
yeah, this should be renamed